### PR TITLE
Enterprise Versioning Compatibility Matrix

### PR DESCRIPTION
**Pull request by paolaperaza**
Starting a PR here that creates a new Versioning Compatibility doc for Enterprise prospects and customers to refer to. A description of what's needed is included in this GH issue: https://github.com/astronomer/issues/issues/1652

Working with @jpweber to finalize the versions currently listed - we may expand this doc from there to include links to third-party upgrade guides etc.

### CODE REVIEW

File: enterprise/v0.16/reference/01_platform_version-compatibility.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+## Overview
+
+The Astronomer Platform ships with and requires a number of adjacent technologies that support it, including Kubernetes, Helm and Apache Airflow itself. For users looking to install or upgrade Astronomer, we've provided a reference of all required tooling with corresponding versions that are compatible with each long-term support (LTS) version of Astronomer Enterprise.
+
+## Astronomer Enterprise Version Compatibility Matrix
+
+| Astronomer Platform | Kubernetes             | Helm | Terraform | Postgres | Astronomer Certified                    | Python |
+|---------------------|------------------------|------|-----------|----------|-----------------------------------------|--------|
+| v0.16               | 1.14, 1.15, 1.16, 1.17 | 3    | 0.12      | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12        | 2, 3   |
+| v0.22               | 1.15, 1.16, 1.17       | 3    | 0.12      | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 2.0.0 | 2, 3   |
```
Comment (kaxil) [inline enterprise/v0.16/reference/01_platform_version-compatibility.md:16]:
```suggestion
| v0.22               | 1.15, 1.16, 1.17       | 3    | 0.12      | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 2.0.0 |    3   |
```

Comment (kaxil) [reply enterprise/v0.16/reference/01_platform_version-compatibility.md:16]: We only Support Py3.6+

Comment (kaxil) [reply enterprise/v0.16/reference/01_platform_version-compatibility.md:16]: I think even for v0.16 of Platform

Comment (paolaperaza) [reply enterprise/v0.16/reference/01_platform_version-compatibility.md]: Roger that! Changed - thanks @kaxil.

**Review by jpweber — APPROVED** [review]

**Review by kaxil — APPROVED** [review]

**Pull request merged**